packages/loader/container-loader/src/container.ts

```diff
@@ -801,6 +810,40 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
 
             this.emit("closed", error);
 
+            if (this.visibilityEventHandler !== undefined) {
```

Comment:
Only change to `Container.close(...)` is removal of `this.removeAllListeners()` call (so future call to `Container.dispose` will still emit "disposed" event)